File: 0150-evaluate-reverse-polish-notation/solution.java
class Solution {
    public int evalRPN(String[] s) {
       //create an stack to store integers
        Stack<Integer> stk=new Stack<>();
        int result=0;
        for(String s1: s){
            // if string is opetators then pop()  2 values and performs num2 operator num1
          
            if(s1.equals("+") ||  s1.equals("-") ||  s1.equals("*") ||  s1.equals("/")  ){
                //to pop 2 elements and perform operration if of all stack me 2 elements toh hona                 //chihita na
                if(stk.size() >=2){
                    int num1=stk.pop();
                    int num2=stk.pop();
                    
                    if(s1.equals("+") )
                      result=num2 + num1;
                    
                    if(s1.equals("-") )
                      result=num2 - num1;
                    
                    if(s1.equals("*") )
                      result=num2 * num1;
                    
                    if(s1.equals("/") )
                      result=num2 / num1;

                    stk.push(result);
                }
            }
            else{
               try{
                  int val=Integer.parseInt(s1);
                  stk.push(val);
               }
               catch(NumberFormatException e){
                
               }
           }
        }
        return stk.peek();
    }
}

File: 0150-evaluate-reverse-polish-notation/solution.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> stk;

        for(string x : tokens){
            if(x == "+" || x == "-" || x == "*" || x == "/"){
                int top = stk.top();
                int second = top;
                stk.pop();
                top = stk.top();
                int first = top;
                stk.pop();

                int result = 0;
                if(x == "+") result = first + second;
                if(x == "-") result = first - second;
                if(x == "*") result = first * second;
                if(x == "/") result = first / second;

                stk.push(result);
            }else{
                int no = stoi(x);
                stk.push(no);
            }
        }
        int top = stk.top();
        return top;
    }
};
